.circleci/config.yml

```diff
@@ -15,24 +15,58 @@ defaults: &defaults
         POSTGRES_USER: postgres
   working_directory: /app
 
+references:
+  attach_workspace: &attach_workspace
+    attach_workspace:
+      at: /app
+  restore_cache: &restore_cache
+    restore_cache:
+      keys:
+        - bundler-dependencies-{{ checksum "Gemfile.lock" }}
+  install_ruby_dependencies: &install_ruby_dependencies
+    run:
+      name: Install ruby dependencies
+      command: bundle install
+  wait_for_db: &wait_for_db
+    run:
+      name: Wait for db
+      command: dockerize -wait tcp://localhost:5432 -timeout 2m
+  create_test_db: &create_test_db
+    run:
+      name: Create test DB
+      command: cd spec/decidim_dummy_app && RAILS_ENV=test bundle exec rails db:create db:schema:load
+  store_artifacts: &store_artifacts
+    store_artifacts:
+      path: /app/spec/decidim_dummy_app/tmp/screenshots
+  store_test_results: &store_test_results
+    store_test_results:
+      path: ~/rspec
+  run_rspec: &run_rspec
+    run:
+      name: Run RSpec
+      command: mkdir ~/rspec && cd decidim-$CIRCLE_JOB && rake
+  format_test_coverage: &format_test_coverage
+    run:
+      name: Format CodeClimate test coverage
+      command: ./cc-test-reporter format-coverage -t simplecov -o coverage/codeclimate.$CIRCLE_JOB.json coverage/.resultset.json
+  persist_coverage_to_workspace: &persist_coverage_to_workspace
+    persist_to_workspace:
+      root: coverage
+      paths:
+        - codeclimate.{{ .Environment.CIRCLE_JOB }}.json
```

Comment:
Seems that this line is failing. This is the documentation: https://circleci.com/docs/2.0/caching/#using-keys-and-templates. But, I'm not sure if the variable name needs to be in camel case format. I'll ask to Code Climate support ...

Reply:
I think the problem is that it's not escaping the ENV variable here, so it's not using the correct file name.

You might need to keep that one for each job definition, which is not *that* bad I guess (after all, we're still gaining a lot)

Reply:
Yes, seems that the `paths` list aren't interpolated unlike the `keys` list that do.


> The paths list uses Glob from Go, and the pattern matches filepath.Match.

Docs: https://circleci.com/docs/2.0/configuration-reference/#example-for-root-key


I've thought about changing `codeclimate.{{ .Environment.CIRCLE_JOB }}.json` to `codeclimate.*.json`, but I'm not sure if this can create problems (I don't fully understand the documentation).


> Each workflow has a temporary workspace associated with it. The workspace can be used to pass along unique data built during a job to other jobs in the same workflow. Jobs can add files into the workspace using the persist_to_workspace step and download the workspace content into their fileystem using the attach_workspace step. The workspace is additive only, jobs may add files to the workspace but cannot delete files from the workspace. Each job can only see content added to the workspace by the jobs that are upstream of it.
> 
> When attaching a workspace the “layer” from each upstream job is applied in the order the upstream jobs appear in the workflow graph. When two jobs run concurrently the order in which their layers are applied is undefined. If multiple concurrent jobs persist the same filename then attaching the workspace will error.


Docs: https://circleci.com/docs/2.0/configuration-reference/#attach_workspace

Reply:
I think it won't be a problem, let's try it and see what happens! 😄